docs/plugins.md

```diff
@@ -135,6 +143,18 @@ function embed(
 
 If you don't want to switch to a different parser, simply return `null` or `undefined`.
 
+### Utility functions
+
+A `util` module from prettier core is considered a private API and is not meant to be consumed by plugins. Instead, the `util-shared` module provides the following limited set of utility functions for plugins:
+
+```ts
+makeString(rawContent: string, enclosingQuote: string, unescapeUnnecessarEscapes: boolean): string;
+getNextNonSpaceNonCommentCharacterIndex(text: string, node: object, options: object): number;
+isNextLineEmptyAfterIndex(text: string, index: number): boolean;
+isNextLineEmpty(text: string, node: object, options: object): boolean;
+mapDoc(doc: object, callback: function): void;
```

Comment:
Do you want to 📖 document these now or in another PR?

Reply:
It might make sense to add some minimal documentation for them now? I'd be happy if you could contribute to that, I'm not too familiar with some of them 😉

Reply:
See also https://github.com/prettier/prettier-php/pull/14/files for a usage example